src/clib/pio_darray.c

```diff
@@ -186,8 +187,9 @@ int PIOc_write_darray_multi(int ncid, const int *vid, int ioid, int nvars, PIO_O
      * points. This is generally faster than the netcdf method of
      * filling the entire array with missing values before overwriting
      * those values later. */
-    if (iodesc->rearranger == PIO_REARR_SUBSET && iodesc->needsfill &&
-        iodesc->holegridsize > 0)
+    if (iodesc->rearranger == PIO_REARR_SUBSET)
+	LOG((2, "pio_write_darray_multi nvars=%d holegridsize=%ld %d\n",nvars, iodesc->holegridsize, iodesc->needsfill));
+    if (iodesc->rearranger == PIO_REARR_SUBSET && iodesc->needsfill)
```

Comment:
bget() assumes requested size > 0 (when not using malloc), so does this change cause issues on procs where holegridsize is 0 ?